tests/rptest/tests/franz_go_verifiable_test.py

```diff
@@ -0,0 +1,155 @@
+            1000, 20, self._node_for_franz_go)
+
+    # In the future producer will signal about json creation
+    def _create_json_file(self):
+        small_producer = FranzGoVerifiableProducer(
+            self.test_context, self.redpanda, self.topic,
+            FranzGoVerifiableTest.MSG_SIZE, 1000, self._node_for_franz_go)
+        small_producer.start()
+        small_producer.wait()
+
+    @cluster(num_nodes=4)
+    def test_with_all_type_of_loads(self):
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        # Need create json file for consumer at first
+        self._create_json_file()
+
+        self._producer.start()
+        self._seq_consumer.start()
+        self._rand_consumer.start()
+
+        self._producer.wait()
+
+
+class FranzGoVerifiableWithSiTest(RedpandaTest):
+    MSG_SIZE = 1000000
+    segment_size = 5 * 1000000
+
+    topics = (TopicSpec(partition_count=100, replication_factor=3), )
+
+    def __init__(self, ctx):
+        si_settings = SISettings(
+            log_segment_size=FranzGoVerifiableWithSiTest.segment_size,
+            cloud_storage_cache_size=5 *
+            FranzGoVerifiableWithSiTest.segment_size)
+
+        super(FranzGoVerifiableWithSiTest, self).__init__(
+            test_context=ctx,
+            num_brokers=3,
+            extra_rp_conf={
+                # Disable prometheus metrics, because we are doing lots
+                # of restarts with lots of partitions, and current high
+                # metric counts make that sometimes cause reactor stalls
+                # during shutdown on debug builds.
+                'disable_metrics': True,
+
+                # We will run relatively large number of partitions
+                # and want it to work with slow debug builds and
+                # on noisy developer workstations: relax the raft
+                # intervals
+                'election_timeout_ms': 5000,
+                'raft_heartbeat_interval_ms': 500,
+            },
+            si_settings=si_settings)
+
+        self._node_for_franz_go = ctx.cluster.alloc(
+            ClusterSpec.simple_linux(1))
+
+        self._producer = FranzGoVerifiableProducer(
+            ctx, self.redpanda, self.topic,
+            FranzGoVerifiableWithSiTest.MSG_SIZE, 20000,
+            self._node_for_franz_go)
+        self._seq_consumer = FranzGoVerifiableSeqConsumer(
+            ctx, self.redpanda, self.topic,
+            FranzGoVerifiableWithSiTest.MSG_SIZE, self._node_for_franz_go)
+        self._rand_consumer = FranzGoVerifiableRandomConsumer(
+            ctx, self.redpanda, self.topic,
+            FranzGoVerifiableWithSiTest.MSG_SIZE, 1000, 20,
+            self._node_for_franz_go)
+
+    def tearDown(self):
+        sef.redpanda.delete_bucket_from_si()
```

Comment:
it feels awkward having the RedpandaService be responsible for building an SI client, creating bucket(s), etc... but then depend on the test harness to clean-up.

I originally thought about suggesting that we should have a test base class for SI tests that would do GC and the setup that you added into the RedpandaService. But it does seem like being able to easily turn on SI within the service for any test makes sense.

So I wonder if we can still do the clean-up in the RedpandaService? I believe this functionality should be added into core ducktape (which we can do), but perhaps an easier thing would be to piggy back on @jcsp's `raise_on_bad_log_line` infrastructure which smells like service-level tear down functionality?

Reply:
There is a Service.clean method (RedpandaService could implement this and call delete_bucket_from_si() before doing a super().clean())

My thinking with `@cluster` (where the call to raise_on_bad_log_line lives) is to use if for checks called as part of the test itself, to make sure that we are surfacing failure properly.  For things that should always work (and aren't redpanda bugs if they don't), I think using clean is okay.